lib/modules/powershell/lateral_movement/invoke_portfwd.py

```diff
@@ -0,0 +1,157 @@
+
+        # The PowerShell script itself, with the command to invoke for
+        #   execution appended to the end. Scripts should output everything
+        #   to the pipeline for proper parsing.
+        #
+        # If you're planning on storing your script in module_source as a ps1,
+        #   or if you're importing a shared module_source, use the first
+        #   method to import it and the second to add any additional code and
+        #   launch it.
+        #
+        # If you're just going to inline your script, you can delete the first
+        #   method entirely and just use the second. The script should be
+        #   stripped of comments, with a link to any original reference script
+        #   included in the comments.
+        #
+        # First method: Read in the source script from module_source
+        moduleSource = self.mainMenu.installPath + "/data/module_source/lateral_movement/Invoke-PortFwd.ps1"
+        if obfuscate:
+            helpers.obfuscate_module(moduleSource=moduleSource, obfuscationCommand=obfuscationCommand)
+            moduleSource = moduleSource.replace("module_source", "obfuscated_module_source")
+        try:
+            f = open(moduleSource, 'r')
+        except:
+            print(helpers.color("[!] Could not read module source path at: " + str(moduleSource)))
+            return ""
+
+        moduleCode = f.read()
+        f.close()
+
+        # If you'd just like to import a subset of the functions from the
+        #   module source, use the following:
+        #   script = helpers.generate_dynamic_powershell_script(moduleCode, ["Get-Something", "Set-Something"])
+        script = moduleCode
+
+        # Second method: For calling your imported source, or holding your
+        #   inlined script. If you're importing source using the first method,
+        #   ensure that you append to the script variable rather than set.
+        #
+        # The script should be stripped of comments, with a link to any
+        #   original reference script included in the comments.
+        #
+        # If your script is more than a few lines, it's probably best to use
+        #   the first method to source it.
+        #
+        # script += """
+
+        scriptEnd = "Invoke-PortFwd"
+
+        # Add any arguments to the end execution of the script
+        for option, values in self.options.items():
+            if option.lower() != "agent":
+                if values['Value'] and values['Value'] != '':
+                    if values['Value'].lower() == "true":
+                        # if we're just adding a switch
+                        scriptEnd += " -" + str(option)
+                    else:
+                        scriptEnd += " -" + str(option) + " " + str(values['Value'])
+        if obfuscate:
+            scriptEnd = helpers.obfuscate(psScript=scriptEnd, installPath=self.mainMenu.installPath, obfuscationCommand=obfuscationCommand)
+        script += scriptEnd
+
```

Comment:
I will have to update the template since this is missing. Can you also please add in a line to call the keyword obfuscation function.

`script = helpers.keyword_obfuscation(script)`

Reply:
And that, as well =)